Tools/clinic/clinic.py

```diff
@@ -1579,7 +1590,7 @@ def render_function(self, clinic, f):
 
 
 @contextlib.contextmanager
-def OverrideStdioWith(stdout):
+def OverrideStdioWith(stdout: io.TextIOWrapper):
```

Comment:
Soooo, I was actually looking at this earlier today. I'm not sure it's worth annotating this because... maybe we should just delete it and use `contextlib.redirect_stdout`? :)

Reply:
All tests pass if I do this, though, which is somewhat concerning:

```diff
--- a/Tools/clinic/clinic.py
+++ b/Tools/clinic/clinic.py
@@ -1580,6 +1580,7 @@ def render_function(self, clinic, f):

 @contextlib.contextmanager
 def OverrideStdioWith(stdout):
+    1/0
     saved_stdout = sys.stdout
     sys.stdout = stdout
     try:
```

Reply:
Make it part of #104683 ! I'll revert that particular commit.